apps/emqx_connector_aggregator/src/emqx_connector_aggreg_delivery.erl

```diff
@@ -151,8 +175,9 @@ process_append_records(
         transfer = Transfer0
     }
 ) ->
-    {Writes, Container} = emqx_connector_aggreg_container:fill(ContainerMod, Records, Container0),
-    Transfer = Mod:process_append(Writes, Transfer0),
+    {Writes, WriteMetadata, Container} =
+        emqx_connector_aggreg_container:fill(ContainerMod, Records, Container0),
+    Transfer = Mod:process_append(Writes, WriteMetadata, Transfer0),
```

Comment:
Design-wise, this looks weird: it's tight coupling between aggregator and transfer implementations. Which means it won't probably work in arbitrary combinations, which in turn defies the purpose of such abstractions.

The issue though is that this module sort of forces us to have 2 different components, and this is evidently wrong. I'd say more clean way is to make this module aware of situations when there's no container and transfers work directly with records (so that `emqx_bridge_s3tables_impl` deals with Avro directly). This perhaps also achievable by supplying "noop-container" as a quick workaround.